homeassistant/components/flux_led/config_flow.py

```diff
@@ -0,0 +1,232 @@
+"""Config flow for Flux LED/MagicLight."""
+import copy
+import logging
+
+from flux_led import BulbScanner
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+from .const import (
+    CONF_AUTOMATIC_ADD,
+    CONF_CONFIGURE_DEVICE,
+    CONF_DEVICES,
+    CONF_EFFECT_SPEED,
+    CONF_REMOVE_DEVICE,
+    DEFAULT_EFFECT_SPEED,
+    DOMAIN,
+    SIGNAL_ADD_DEVICE,
+    SIGNAL_REMOVE_DEVICE,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for FluxLED/MagicHome Integration."""
+
+    VERSION = 1
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry: config_entries.ConfigEntry):
+        """Get the options flow for the Flux LED component."""
+        return OptionsFlow(config_entry)
+
+    async def async_step_import(self, data: dict = {}):
+        """Handle configuration via YAML import."""
+        _LOGGER.debug("Importing configuration from YAML for flux_led")
+        config_entry = self.hass.config_entries.async_entries(DOMAIN)
+
+        _LOGGER.warning(
+            "Your flux_led configuration has already been imported. Please remove it from your configuration"
+        )
+   
+        if config_entry:
+            return self.async_abort(reason="single_instance_allowed")
+
+        return await self.async_step_user(
+            user_input={
+                CONF_AUTOMATIC_ADD: data[CONF_AUTOMATIC_ADD],
+                CONF_DEVICES: data.get(CONF_DEVICES, {}),
+            }
+        )
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
```

Comment:
The user may want to enter the up address manually here.

Examples in both Yeelight and tplink